packages/ui/src/EditTrackCard.tsx

```diff
@@ -2,23 +2,46 @@
 import { useState } from "react";
 import { Button } from "./shad/ui/button";
 import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "./shad/ui/card";
-import { Track } from "@prisma/client";
+import { Track,Categories } from "@prisma/client";
 import { Input } from "./shad/ui/input";
 import { updateTrack } from "../../../apps/web/components/utils";
 
-const EditTrackCard = ({ Track }: { Track: Track }) => {
+interface TrackCardProps extends Track {
+  categories: {
+    category: {
+      id: string;
+      category: string;
+    };
+  }[];
+}
+
+const EditTrackCard = ({ Track,categories }: { 
+  Track: TrackCardProps 
+  categories: Categories[] }) => {
   const [isEditing, setIsEditing] = useState(false);
   const [title, setTitle] = useState(Track.title);
   const [description, setDescription] = useState(Track.description);
   const [image, setImage] = useState(Track.image);
   const [hidden, setHidden] = useState(Track.hidden);
+  const [selectedCategory, setSelectedCategory] = useState<string[]>(Track.categories.map((item) => item.category.id));
+
+
   function handleEdit(id: string) {
     if (isEditing) {
-      updateTrack(id, { title, description, image, hidden });
+      updateTrack(id, { id, title, description, image, hidden, selectedCategory});
```

Comment:
Would this work?
`selectedCategory` isn't a key in the `db.track.update` function

Have u tested to see if this works?

Reply:
it is key `const [selectedCategory, setSelectedCategory] = useState<string[]>(Track.categories.map((item) => item.category.id));`

Reply:
> Have u tested to see if this works?

ya I have tested it

Reply:
Can u update the type of the `updateTrack` function from any to the right type as well?
Over here - https://github.com/code100x/daily-code/blob/main/apps/web/components/utils.tsx#L62

Reply:
yes i will

Reply:
> Can u update the type of the `updateTrack` function from any to the right type as well? Over here - https://github.com/code100x/daily-code/blob/main/apps/web/components/utils.tsx#L62

it's done

Reply:
hey, @hkirat 
I have fixed this https://github.com/code100x/daily-code/pull/398#issuecomment-2099042050 
the import error in `LessonView.tsx`
in a commit https://github.com/code100x/daily-code/pull/400/commits/75fd7b381ea64d0fd086786d0d6f13c809897886